internal/services/connection_check.go

```diff
@@ -159,7 +161,10 @@ func (cs *connectionService) connectionCheck() {
 			"brokerid":  strconv.Itoa(int(b.ID())),
 			"connected": strconv.FormatBool(connected),
 		}
-
+		// initialize all error-related metrics with starting value of 0
+		if !cs.initialized {
```

Comment:
I cannot understand why we need an "initialization" check here (so using a boolean flag) instead of initializing the `connectionError` in the `NewConnectionService` when it's created for the first time. What am I missing here?

Reply:
@ppatierno it was due to the addition of labels ("brokerid", "connected") that are instantiated after attempting to connect.

Reply:
I would avoid this logic, what about this ...
When it's connected (so the `if connected`) we can call the `connectionError.With(labels).Add(0)`. It means that if it's the first time and connects, initialize the metric to 0; if it's not the first time, adding 0 doesn't change the previous value. Instead if you have an error, you are on the other condition and `connectionError.With(labels).Inc()` is executed.
Could it work? Wdyt?

Reply:
you are right. is committed